src/main/java/com/openshift/jenkins/plugins/pipeline/dsl/OpenShiftBuilder.java

```diff
@@ -153,6 +160,22 @@ public Step newInstance(Map<String, Object> arguments) throws Exception {
                     step.setWaitTime(waitTime.toString());
                 }
             }
+
+             // Allow env to be specified as a map: env: [ name : 'value1', 'name2' : 'value2' ]
+            Object envObject = arguments.get( "env" );
+            if ( envObject != null ) {
+                if ( envObject instanceof Map ) {
+                    Map map = (Map)envObject;
+                    Iterator<Map.Entry> it = map.entrySet().iterator();
+                    while ( it.hasNext() ) {
+                        Map.Entry entry = it.next();
+                        step.getEnvVars().add( new NameValuePair( entry.getKey().toString(), entry.getValue().toString() ) );
+                    }
+                } else {
+                    LOGGER.log(Level.WARNING, "Environment variables must be specified as a map. Received: " + envObject.getClass().getName() );
```

Comment:
throwing an exception of some sort (i don't have strong preference between a runtime exception and a custom built exception) seems prudent here so that the error shows up in the job log; otherwise, the job may succeed without the env vars he was expecting and he may not think to double check the jenkins log